src/app/files.service.ts

```diff
@@ -100,21 +134,25 @@ export class FilesService {
                 this.httpGETRequest.unsubscribe();
             }
             this.httpGETRequest = this.http
-                .get(this.configService.getURL('files/local' + filePath), this.configService.getHTTPHeaders())
+                .get(this.configService.getURL('files' + filePath), this.configService.getHTTPHeaders())
                 .subscribe(
                     (data: OctoprintFilesAPI): void => {
-                        let filamentLength = 0;
-                        _.forEach(data.gcodeAnalysis.filament, (tool): void => {
-                            filamentLength += tool.length;
-                        });
+                        if (data.gcodeAnalysis) {
+                            var filamentLength = 0;
```

Comment:
```suggestion
                        let filamentLength = 0;
                        if (data.gcodeAnalysis) {
```

Is there a need for filamentLength to be function scoped?

Reply:
good point, fixed both occurrences